andino_firmware/src/motor.h

```diff
@@ -62,47 +62,44 @@
 // CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 // OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#ifndef ANDINO_FIRMWARE_SRC_MOTOR_H_
+#define ANDINO_FIRMWARE_SRC_MOTOR_H_
 
-#include "motor_driver.h"
+namespace andino {
 
-#include "Arduino.h"
-#include "commands.h"
-#include "hw.h"
+class Motor {
+ public:
+  /// @brief Constructs a new Motor object.
+  ///
+  /// @param enable_gpio_pin Motor enable GPIO pin.
+  /// @param forward_gpio_pin Motor forward GPIO pin.
+  /// @param backward_gpio_pin Motor backward GPIO pin.
+  Motor(int enable_gpio_pin, int forward_gpio_pin, int backward_gpio_pin)
+      : enable_gpio_pin_(enable_gpio_pin),
+        forward_gpio_pin_(forward_gpio_pin),
+        backward_gpio_pin_(backward_gpio_pin) {}
```

Comment:
Should we support not having the `enable_gpio_pin` here? Not sure if std::optional(cpp17) is available here.(doubt it)

Reply:
Good point. AFAIK, AVR boards are compiled with -std=c++11 and they lack libc++ and STL.

I agree with you that supporting not having to set `enable_gpio_pin` would be a nice enhancement for this class, but I'm a bit concerned that we would be adding coding that we won't be using nor testing (and therefore might easily get rotten that way). Given that trying to add unit tests is on the roadmap, I propose to revisit this once we have been able to create a test to check the corresponding code flow. WFYT?

Reply:
LGTM